### PR TITLE
Make error exposure nagios binding configurable

### PR DESCRIPTION
**Pull request by Peterburnett**

### CODE REVIEW

File: lang/en/tool_heartbeat.php [hunk 1]
```diff
@@ -52,6 +52,11 @@
 $string['checkrangerequestok'] = 'Range requests are working, 206 response with only 10 bytes of data';
 $string['checkrangerequestbad'] = 'Range requests are bad! HTTP {$a->code} response with only {$a->bytes} bytes of data for {$a->url}';
 $string['ips_combine'] = 'The IPs listed above will be combined with the IPs listed below.';
+$string['errorascritical'] = ' Report check errors as:';
+$string['errorascritical_desc'] = 'This setting controls what check API errors are reported as in Nagios. "CRITICAL" is the most noisy, and "WARNING" is the least noisy for monitoring endpoints. Business hours is 9AM - 5PM in the server timezone.';
+$string['error_critical'] = 'CRITICAL';
```
Comment (brendanheywood) [inline lang/en/tool_heartbeat.php:57]: Are these strings in core?

File: croncheck.php [hunk 2]
```diff
@@ -299,7 +299,18 @@
...
+            // If we have an error, how should we handle it.
+            if ($status == \core\check\result::ERROR && !$critical) {
+                $mapping = get_config('tool_heartbeat', 'errorcritical');
+                if ($mapping === 'critical') {
+                    $critical = true;
+                } else if ($mapping === 'criticalbusiness') {
+                    // Here we should only set the critical flag between 0900 and 1700 server time.
+                    $time = new DateTime('now', core_date::get_server_timezone_object());
+                    $hour = (int) $time->format('H');
+                    $critical = ($hour >= 9 && $hour <= 17);
```
Comment (brendanheywood) [inline croncheck.php:311]: <= 17 here mean it will be critical up to 5:59 pm

File: lang/en/tool_heartbeat.php [hunk 3]
```diff
@@ -52,6 +52,11 @@
 $string['checkrangerequestok'] = 'Range requests are working, 206 response with only 10 bytes of data';
 $string['checkrangerequestbad'] = 'Range requests are bad! HTTP {$a->code} response with only {$a->bytes} bytes of data for {$a->url}';
 $string['ips_combine'] = 'The IPs listed above will be combined with the IPs listed below.';
+$string['errorascritical'] = ' Report check errors as:';
+$string['errorascritical_desc'] = 'This setting controls what check API errors are reported as in Nagios. "CRITICAL" is the most noisy, and "WARNING" is the least noisy for monitoring endpoints. Business hours is 9AM - 5PM in the server timezone.';
```
Comment (brendanheywood) [inline lang/en/tool_heartbeat.php:56]: Can you include the timezone value in help here so it’s obvious if it’s wrong of if the server is our timezone but not the clients timezone

File: croncheck.php (hunk 2 above)
Comment (Peterburnett) [reply croncheck.php:311]: Good catch

File: lang/en/tool_heartbeat.php (hunk 1 above)
Comment (Peterburnett) [reply lang/en/tool_heartbeat.php:57]: Not in the all caps case, and I wanted to make it clear that this directly to the nagios code

**Pull request merged**